### PR TITLE
Add missing client list_ methods.

### PR DESCRIPTION
**Pull request by agorajek**
### Summary

1. Added all remaining list_* methods to the Python client.
2. Added a special error message when creating cube node. Because the current error may be not enough / confusing.

### Test Plan

unit tests

- [x] PR has an associated issue: #630 
- [x] `make check` passes
- [x] `make test` shows 100% unit test coverage

### CODE REVIEW

File: datajunction-clients/python/datajunction/_internal.py [hunk 1]
```diff
@@ -203,13 +203,25 @@ def process_results(results) -> "pd.DataFrame":
...
+
+    def _get_all_nodes(
+        self,
+        type_: Optional[models.NodeType] = None,
+    ):
+        """
+        Retrieve all nodes of a given type.
+        """
+        response = self._session.get(
+            "/nodes/" + (f"?node_type={type_.value}" if type_ else ""),
```
Comment (samredai) [inline datajunction-clients/python/datajunction/_internal.py:224]: This is one of two ways to get a list of nodes of a specific type--there's also the /metrics and /dimensions endpoints. I wonder if we should get rid of those so there's just a single way to do this. On a similar note, I wonder if this /nodes endpoint should also have an optional `?namespace=...` argument and then we can get rid of the /namespaces/{namespace}/ endpoint. Not saying this all needs to be done in this PR but curious to hear your thoughts on it.

**Review by samredai — APPROVED** [review]
LGTM, just had a comment that's more about how the server API is laid out.

File: datajunction-clients/python/datajunction/_internal.py (hunk 1 above)
Comment (agorajek) [reply datajunction-clients/python/datajunction/_internal.py:224]: Totally agree with simplifying the server side. We don't need separate listing methods for each node type on the server side. I was actually going to do it in the same PR, just ran out of time. Will do in a follow-up.

Comment (agorajek) [reply datajunction-clients/python/datajunction/_internal.py:224]: https://github.com/DataJunction/dj/issues/716

File: datajunction-clients/python/datajunction/client.py [hunk 2]
```diff
@@ -31,31 +29,76 @@ def list_namespaces(self, prefix: Optional[str] = None) -> List[str]:
...
+            return self._get_nodes_in_namespace(
+                namespace=namespace,
+                type_=models.NodeType.CUBE,
+            )
+        return self._get_all_nodes(type_=models.NodeType.CUBE)
+
+    #
+    # List other nodes: sources, transforms, all.
+    #
+    def list_sources(self, namespace: Optional[str] = None) -> List[str]:
```
Comment (shangyian) [inline datajunction-clients/python/datajunction/client.py:66]: Should there also be an endpoint to list source nodes that were created via the register-a-table API endpoint? I know that users can also do that just by calling `list_sources(namespace="source")` (or whatever the default namespace was configured to be server-side), but they might not know what the default registered table sources namespace is. 

No need to do it in this PR, of course!

**Review by shangyian — APPROVED** [review]
LGTM!

**Pull request merged**